src/coreclr-debug/debugConfigurationProvider.ts

```diff
@@ -59,6 +61,42 @@ export class DotnetDebugConfigurationProvider implements vscode.DebugConfigurati
                 return undefined;
             }
         }
+        
+        if (!this.platformInformation.isLinux() && !vscode.env.remoteName && vscode.env.uiKind != vscode.UIKind.Web) 
+        {
+            if(debugConfiguration.checkForDevCert === undefined && debugConfiguration.serverReadyAction && !debugConfiguration.pipeTransport)
+            {
+                debugConfiguration.checkForDevCert = true;
+            }
+
+            if (debugConfiguration.checkForDevCert)
+            {
+                hasDotnetDevCertsHttps(this.options.dotNetCliPaths).then(async (hasDevCert) => {
+                    if(!hasDevCert)
+                    {
+                        const result = await vscode.window.showInformationMessage(
+                            "The selected launch configuration is configured to launch a web browser but no trusted development certificate was found. Create a trusted self-signed certificate?", 
+                            "Don't Ask Again", 'Not Now', 'Yes'
+                            ); 
+                        if (result === 'Yes')
+                        {
+                            if (await createSelfSignedCert(this.options.dotNetCliPaths))
+                            {
+                                vscode.window.showInformationMessage('Self-signed certificate sucessfully created');
+                            }
+                            else
+                            {
+                                vscode.window.showWarningMessage("Couldn't create self-signed certificate");
+                            }
+                        }
+                        else if (result === "Don't Ask Again")
+                        {
+                            await vscode.window.showInformationMessage("To don't see this message again set launch option `checkForDevCert` to `false` in launch.json", { modal: true });
+                        }
```

Comment:
Rather than having a button for this, I would just stick this as part of the message (unless VS Code has a max length).

Reply:
The problem is VS Code messages [don't allow formatting or break lines in them](https://github.com/microsoft/vscode/issues/48900), so it makes it not very readable to add that that information in one big block of text.

Reply:
Got it. Do you know if we do anything similar anywhere else? I wonder if we want a 'More information' button instead, and we could take you to the debugger-launchjson.md section for this option.

CC @sayedihashimi in case he has any thoughts

Reply:
I should add, the reason I am suggesting this is that I feel like have the button be called "Don't Ask Again" rather over promises.

Reply:
We could use the [`detail`](https://github.com/microsoft/vscode/blob/c83eff40dd2c2115f5fd5e67a6ce304fbc6fdefa/src/vscode-dts/vscode.d.ts#L1974) property to provide this information.

Reply:
> We could use the [`detail`](https://github.com/microsoft/vscode/blob/c83eff40dd2c2115f5fd5e67a6ce304fbc6fdefa/src/vscode-dts/vscode.d.ts#L1974) property to provide this information.

For this the main info message would need be to be a modal which might be to obstructing. 

> Got it. Do you know if we do anything similar anywhere else? I wonder if we want a 'More information' button instead, and we could take you to the debugger-launchjson.md section for this option.
> 
> CC @sayedihashimi in case he has any thoughts

I'm liking this idea.

Reply:
> Got it. Do you know if we do anything similar anywhere else? I wonder if we want a 'More information' button instead, and we could take you to the debugger-launchjson.md section for this option.

Implemented this

Reply:
I like the idea where we show a More Info link.